alacritty_terminal/tests/ref/grid_reset/size.json

```diff
@@ -1 +1 @@
-{"width":2532.0,"height":1380.0,"cell_width":9.0,"cell_height":19.0,"padding_x":3.0,"padding_y":3.0}
\ No newline at end of file
+{"width":1262.0,"height":1532.0,"cell_width":18.0,"cell_height":35.0,"padding_x":0.0,"padding_y":0.0,"dpr":2.0}
```

Comment:
Can you tell me why this reftest was re-recorded? Seems a bit strange to me.

Reply:
It finds out that inner storage differs now... So the failure is a bit strange.
```
failures:

---- grid_reset stdout ----
thread 'grid_reset' panicked at 'Ref test failed; grid doesn't match', alacritty_terminal/tests/ref.rs:141:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.


failures:
    grid_reset

test result: FAILED. 33 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out

```

However with some debug output I see.
```
term_grid: len: 172, num_cols: 280
grid: len: 0, num_cols: 280
```
I personally don't quite understand this `grid: len: 0 thing`, but the 172 kind of make sense.

Reply:
We should find out what exactly is going on here, before doing anything else. Just re-recording reftests without understanding why they fail is not a good idea at all.

I'd assume this might be related to the original reason for initializing everything.

Reply:
So basically the "problem" is that we're actually can call `truncate` now, since on the current master `self.inner.len()` and `self.len()` are identical, and a stupid impl when you allocate lines without some internal  step (like on this PR right) buffer you will also follow the current master behavior.

However on the current PR you can call `truncate` in `shrink_lines` quite easily now. The implementation of `truncate` seems legit and I don't get why leaving old content on a grid fixes it. There's nothing bad with it tbh, since you're likely reuse it afterwards, and it shouldn't change anything, but maybe I'm wrong...

During the test we're calling `initialize`, `rotate`, `shrink_lines`, and a `truncate`, so the problem could be in a one of those functions.